exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/QueryManager.java

```diff
@@ -280,8 +281,15 @@ public void interrupted(final InterruptedException ex) {
     }
   }
 
-  QueryState updateEphemeralState(final QueryState queryState) {
-    switch (queryState) {
+  void updateEphemeralState(final QueryState queryState) {
+      // If query is already in zk transient store, ignore the transient state update option.
+      // Else, they will not be removed from transient store upon completion.
+      if (transientProfiles.get(stringQueryId) == null &&
```

Comment:
Why not just check the option?

`transientProfiles.get(stringQueryId)` is quite expensive itself ([contacts ZooKeeper and deserializes data](https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/coord/zk/ZkEphemeralStore.java#L61)).